### PR TITLE
ASE: ported FPGA property related unit tests

### PR DESCRIPTION
**Pull request by lzhan55**

### CODE REVIEW

**Comment by lzhan55** [comment]
image

**Comment by lzhan55** [comment]
These tests were mostly ported from opae-c/test_props_c.cpp with several additional tests added.

File: testing/ase/test_props_c_ase.cpp [hunk 1]
```diff
@@ -0,0 +1,3360 @@
...
+  fpga_token parent = nullptr;
+
+  ASSERT_EQ(fpgaGetProperties(NULL, &prop), FPGA_OK);
+  EXPECT_EQ(fpgaPropertiesSetObjectType(prop, FPGA_DEVICE), FPGA_OK);
+  EXPECT_EQ(fpgaClearProperties(prop), FPGA_OK);
+
+  // now get the parent token from the prop structure
+  EXPECT_EQ(fpgaPropertiesSetParent(prop, tok_dev_), FPGA_OK);
+  // now get the parent token from the prop structure
+  EXPECT_EQ(fpgaPropertiesGetParent(prop, &parent), FPGA_OK);
```
Comment (ibarraza) [inline testing/ase/test_props_c_ase.cpp]: parent doesn't get destroyed.

File: testing/ase/test_props_c_ase.cpp [hunk 2]
```diff
@@ -0,0 +1,3360 @@
...
+TEST_F(properties_c_ase_p, get_guid01) {
+  fpga_properties prop = NULL;
+  fpga_result result = fpgaGetProperties(NULL, &prop);
+
+  ASSERT_EQ(result, FPGA_OK);
+  ASSERT_FALSE(NULL == prop);
+
+  struct _fpga_properties* _prop = (struct _fpga_properties*)prop;
+
+  // set the object type an
```
Comment (ibarraza) [inline testing/ase/test_props_c_ase.cpp]: src doesn't get destroyed.

File: testing/ase/test_props_c_ase.cpp (hunk 1 above)
Comment (lzhan55) [reply testing/ase/test_props_c_ase.cpp]: good catch! thanks.

File: testing/ase/test_props_c_ase.cpp (hunk 2 above)
Comment (lzhan55) [reply testing/ase/test_props_c_ase.cpp]: Yes, and will fix it. Thanks.

**Pull request merged**